### PR TITLE
i18n(zh-cn): Translate islands, fonts, docsearch, nav & ui files

### PR DESCRIPTION
**Pull request by zihan-ch**
#### What kind of changes does this PR include?

- Translated content

#### Description
add islands.md translation
change and add some the UI strings

### CODE REVIEW

File: src/pages/zh-cn/concepts/islands.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Astro群岛
+description: "Astro群岛（又称组件群岛）是Astro开创的一种新Web架构模式。“群岛架构”由Etsy前端架构工程师Katie Sylor-Miller于2019年首次提出，并由Preact的作者Jason Miller进行扩展。"
```
Comment (kecrily) [inline src/pages/zh-cn/concepts/islands.md:4]:
```suggestion
title: Astro 群岛
description: "Astro 群岛（又称组件群岛）是 Astro 开创的一种新 Web 架构模式。“群岛架构”由 Etsy 前端架构工程师 Katie Sylor-Miller 于2019年首次提出，并由 Preact 的作者 Jason Miller 进行扩展。"
```

File: src/pages/zh-cn/guides/fonts.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    h1 {
+      font-family: "Twinkle Star", cursive;
+    }
+    ```
+
+## 更多资源
+
+### 使用 Tailwind 添加字体
+
+如果您使用的是 [Tailwind 集成](/zh-cn/guides/integrations-guide/tailwind/)，您可以直接为本地字体添加`@font-face`，或使用 Fontsource 的`import`策略来注册您的字体，然后按照 Tailwind 的文档添加自定义字体。
```
Comment (kecrily) [inline src/pages/zh-cn/guides/fonts.md:110]:
```suggestion
如果您使用的是 [Tailwind 集成](/zh-cn/guides/integrations-guide/tailwind/)，您可以直接为本地字体添加 `@font-face`，或使用 Fontsource 的 `import` 策略来注册您的字体，然后按照 Tailwind 的文档添加自定义字体。
```

File: src/pages/zh-cn/guides/fonts.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    导入后Astro并添加`@font-face`以设置字体所需的必要属性。
+
+    ```astro
+    ---
+    // src/layouts/BaseLayout.astro
+    import '@fontsource/twinkle-star';
+    ---
+    ```
+
+4. 使用该`font-family`字体的 Fontsource 页面上显示的。这将适用于可以在 Astro 项目中编写 CSS 的任何地方。
```
Comment (kecrily) [inline src/pages/zh-cn/guides/fonts.md:98]:
```suggestion
4. 使用该 `font-family` 字体的 Fontsource 页面上显示的。这将适用于可以在 Astro 项目中编写 CSS 的任何地方。
```

File: src/pages/zh-cn/guides/fonts.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      </Fragment>
+      <Fragment slot="yarn">
+      ```shell
+      yarn add @fontsource/twinkle-star
+      ```
+      </Fragment>
+    </PackageManagerTabs>
+
+    :::tip
+    您可以在 Fontsource 上每个字体页面的“快速安装”部分找到正确的包名称。包名总是`@fontsource/`以字体名称开头。
```
Comment (kecrily) [inline src/pages/zh-cn/guides/fonts.md:84]:
```suggestion
    你可以在 Fontsource 上每个字体页面的“快速安装”部分找到正确的包名称。包名总是以 `@fontsource/` 字体名称开头。
```

File: src/pages/zh-cn/guides/fonts.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    :::
+
+3. 在要使用字体的布局或组件中导入字体包。通常，您需要在一个通用的布局组件中执行此操作，以确保该字体在您的站点中可用。
+
+    导入后Astro并添加`@font-face`以设置字体所需的必要属性。
```
Comment (kecrily) [inline src/pages/zh-cn/guides/fonts.md:89]:
```suggestion
3. 在要使用字体的布局或组件中导入字体包。通常，需要在通用布局组件中执行此操作，以确保该字体在你的站点中可用。

    导入后 Astro 并添加 `@font-face` 以设置字体所需的必要属性。
```

File: src/pages/zh-cn/guides/fonts.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+      font-family: 'CoolFont', sans-serif;
+    }
+    </style>
+    ```
+
+## 使用字体资源包
+
+[Fontsource](https://fontsource.org/)（字体资源包）项目简化了使用 Google Fonts 和其他开源字体的过程。当你想用什么字体时，只需安装相应的npm模块并稍作配置即可。
+
+1. 从[Fontsource的字体资源目录](https://fontsource.org/fonts)中找到你想用的字体。比如在这个例子中，我们将选择[Twinkle Star](https://fontsource.org/fonts/twinkle-star)字体。
```
Comment (kecrily) [inline src/pages/zh-cn/guides/fonts.md:61]:
```suggestion
1. 从 [Fontsource 的字体资源目录](https://fontsource.org/fonts)中找到你想用的字体。比如在这个例子中，我们将选择 [Twinkle Star](https://fontsource.org/fonts/twinkle-star)字体。
```

File: src/pages/zh-cn/guides/fonts.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+      font-style: normal;
+      font-display: swap;
+    }
+    ```
+
+    :::note
+    我们将`/public`在字体的源URL中，因为所有在`/public`目录下的文件都会被添加到您网站的根目录中。
+    :::
+
+3. 使用`font-family`为组件或布局中的元素设置字体样式。在此示例中，`<h1>`标题将被应用自定义字体，而段落`<p>`不会。
```
Comment (kecrily) [inline src/pages/zh-cn/guides/fonts.md:39]:
```suggestion
3. 使用 `font-family` 为组件或布局中的元素设置字体样式。在此示例中，自定义字体将应用于 `<h1>` 标题，而不会用于段落 `<p>`。
```

File: src/pages/zh-cn/guides/fonts.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+      font-family: 'CoolFont';
+      src: url('/fonts/炫酷字体.woff') format('woff');
+      font-weight: normal;
+      font-style: normal;
+      font-display: swap;
+    }
+    ```
+
+    :::note
+    我们将`/public`在字体的源URL中，因为所有在`/public`目录下的文件都会被添加到您网站的根目录中。
```
Comment (kecrily) [inline src/pages/zh-cn/guides/fonts.md:36]:
```suggestion
    我们将 `/public` 在字体的源链接中，因为所有在 `/public` 目录下的文件都会被添加到网站的根目录中。
```

File: src/pages/zh-cn/guides/fonts.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+如果你想要添加一些自定义字体文件，我们推荐你将它们添加至[`public/` 目录](/zh-cn/core-concepts/project-structure/#public)。接下来你便可以在 CSS 中使用`@font-face`语句注册字体，并使用该`font-family`属性设置页面的字体。
+
+### 例子
+
+想象如果你有一个字体文件`炫酷字体.woff`。
+
+1. 首先将你的自填添加到`public/fonts/`。
+
+2. 在你的 CSS 中添加一个`@font-face`语句。这可能位于你导入的全局`.css`文件中，也可能位于你要使用此字体的布局或组件中的`<style>`块中。
```
Comment (kecrily) [inline src/pages/zh-cn/guides/fonts.md:22]:
```suggestion
想象如果你有一个字体文件 `炫酷字体.woff`。

1. 首先将你的自填添加到 `public/fonts/`。

2. 在你的 CSS 中添加一个 `@font-face` 语句。这可能位于你导入的全局 `.css` 文件中，也可能位于你要使用此字体的布局或组件中的 `<style>` 块中。
```

File: src/pages/zh-cn/guides/fonts.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+layout: ~/layouts/MainLayout.astro
+setup: |
+    import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro';
+---
+
+Astro支持所有常见的在网站中添加自定义字体的方法。本指南将向您展示在项目中包含Web字体的两种不同选项。
+
+## 使用本地字体文件
+
+如果你想要添加一些自定义字体文件，我们推荐你将它们添加至[`public/` 目录](/zh-cn/core-concepts/project-structure/#public)。接下来你便可以在 CSS 中使用`@font-face`语句注册字体，并使用该`font-family`属性设置页面的字体。
```
Comment (kecrily) [inline src/pages/zh-cn/guides/fonts.md:14]:
```suggestion
如果你想要添加一些自定义字体文件，我们推荐你将它们添加至 [`public/` 目录](/zh-cn/core-concepts/project-structure/#public)。接下来你便可以在 CSS 中使用 `@font-face` 语句注册字体，并使用该 `font-family` 属性设置页面的字体。
```

File: src/pages/zh-cn/guides/fonts.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
+---
+title: 使用自定义字体
+description: 想要为Astro生成的网站添加一些自定义的字体？使用Google Fonts或者使用本地文件以添加自定义字体。
+i18nReady: true
+layout: ~/layouts/MainLayout.astro
+setup: |
+    import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro';
+---
+
+Astro支持所有常见的在网站中添加自定义字体的方法。本指南将向您展示在项目中包含Web字体的两种不同选项。
```
Comment (kecrily) [inline src/pages/zh-cn/guides/fonts.md:10]:
```suggestion
Astro 支持所有常见的在网站中添加自定义字体的方法。本指南将向您展示在项目中包含 Web 字体的两种不同选项。
```

File: src/pages/zh-cn/guides/fonts.md [hunk 12]
```diff
@@ -0,0 +1,118 @@
+---
+title: 使用自定义字体
+description: 想要为Astro生成的网站添加一些自定义的字体？使用Google Fonts或者使用本地文件以添加自定义字体。
+i18nReady: true
```
Comment (kecrily) [inline src/pages/zh-cn/guides/fonts.md:4]:
```suggestion
description: 想要为 Astro 生成的网站添加一些自定义的字体？使用 Google Fonts 或者使用本地文件以添加自定义字体。
```

File: src/pages/zh-cn/concepts/islands.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+
+Astro群岛的最明显的好处就是性能：你网站的大部分区域都被转换为了快速、静态的HTML，JavaScript只为单独组件而被加载。JavaScript是一个加载得最慢的资源。每一个字节都影响着阅读者的体验！
+
+另一个好处是并行加载。在上面的一些假想例子中，重要性更低的图像轮播不应该阻挡更重要的页头部分的加载。这两样东西被并行加载并被分别单独组建，这表明阅读者并不需要等着更沉重的图像轮播加载完毕就可以与页头交互了。
+
+还有更棒的：你可以准确地告诉Astro如何以及何时渲染每个组件。如果该图像轮播的加载成本真的很高，你可以附加一个特殊的客户端指令，告诉Astro仅在轮播在页面上可见时才加载它。如果用户从未看到它，它永远不会被加载。
+
+在Astro中，作为开发人员，你可以明确告诉Astro你的页面上的哪些组件也需要在客户端浏览器中运行。Astro 只会准确地补充页面上需要交互性的内容，并将您网站的其余部分保留为静态 HTML。
+
+**群岛正式Astro默认迅速的性能之密码！**
```
Comment (kecrily) [inline src/pages/zh-cn/concepts/islands.md:68]:
```suggestion
Astro 群岛的最明显的好处就是性能：你网站的大部分区域都被转换为了快速、静态的 HTML，JavaScript 只为单独组件而被加载。JavaScript 是一个加载得最慢的资源。每一个字节都影响着阅读者的体验！

另一个好处是并行加载。在上面的一些假想例子中，重要性更低的图像轮播不应该阻挡更重要的页头部分的加载。这两样东西被并行加载并被分别单独组建，这表明阅读者并不需要等着更沉重的图像轮播加载完毕就可以与页头交互了。

还有更棒的：你可以准确地告诉 Astro 如何以及何时渲染每个组件。如果该图像轮播的加载成本真的很高，你可以附加一个特殊的客户端指令，告诉 Astro 仅在轮播在页面上可见时才加载它。如果用户从未看到它，它永远不会被加载。

在 Astro 中，作为开发人员，你可以明确告诉 Astro 你的页面上的哪些组件也需要在客户端浏览器中运行。Astro 只会准确地补充页面上需要交互性的内容，并将您网站的其余部分保留为静态 HTML。

**群岛正是 Astro 在默认情况下能够如此迅速之秘诀！**
```

File: src/pages/zh-cn/concepts/islands.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+```astro title="src/pages/index.astro" ins="client:load"
+---
+// 例子：在此页面上使用一个动态React组件
+---
+<!-- 现在这个组件是可交互性的了！
+  网站的其他部分任然是静态、没有JavaScript的。 -->
+<MyReactComponent client:load />
+```
+
+使用Astro群岛，你的大部分页面保持着纯正、轻盈的HTMl和CSS。在上面的例子中，你紧紧添加了一个简单的、孤立的**可响应岛屿**，而并没有改变任何页面其他部分的代码。
```
Comment (kecrily) [inline src/pages/zh-cn/concepts/islands.md:56]:
```suggestion
但是有些时候，响应式的 UI 是需要客户端 JavaScript 的。你不该将整个页面做成一个像 SPA（单页面应用）一样的 JavaScript 应用，相反，Astro 允许你创建岛屿。

```astro title="src/pages/index.astro" ins="client:load"
---
// 例子：在此页面上使用一个动态 React 组件
---
<!-- 现在这个组件是可交互性的了！
  网站的其他部分任然是静态、没有JavaScript的。 -->
<MyReactComponent client:load />
```

使用 Astro 群岛，你的大部分页面保持着纯正、轻盈的HTMl和CSS。在上面的例子中，你紧紧添加了一个简单的、孤立的**可响应岛屿**，而并没有改变任何页面其他部分的代码。
```

File: src/pages/zh-cn/guides/fonts.md [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+
+### 使用 Tailwind 添加字体
+
+如果您使用的是 [Tailwind 集成](/zh-cn/guides/integrations-guide/tailwind/)，您可以直接为本地字体添加`@font-face`，或使用 Fontsource 的`import`策略来注册您的字体，然后按照 Tailwind 的文档添加自定义字体。
+
+### 了解 Web 字体如何运作
+
+[MDN的 Web 字体指南](https://developer.mozilla.org/en-US/docs/Learn/CSS/Styling_text/Web_fonts) 介绍了这个话题。
+
+### 为你的字体生产 Css
```
Comment (kecrily) [inline src/pages/zh-cn/guides/fonts.md:116]:
```suggestion
### 为你的字体生成 CSS
```

**Comment by delucis** [comment]
Thanks for reviewing @kecrily!

@zihan-ch You should be able to go through Percy's comments one by one and click <kbd>Commit suggestion</kbd> to accept it or comment below to discuss things.

**Comment by zihan-ch** [comment]
> Thanks for reviewing @kecrily!
> 
> @zihan-ch You should be able to go through Percy's comments one by one and click Commit suggestion to accept it or comment below to discuss things.

Ok. I will check it.

**Pull request merged**